src/main/java/com/commercetools/sync/inventory/utils/InventoryUpdateActionUtils.java

```diff
@@ -0,0 +1,109 @@
+package com.commercetools.sync.inventory.utils;
+
+import io.sphere.sdk.channels.Channel;
+import io.sphere.sdk.commands.UpdateAction;
+import io.sphere.sdk.inventory.InventoryEntry;
+import io.sphere.sdk.inventory.InventoryEntryDraft;
+import io.sphere.sdk.inventory.commands.updateactions.ChangeQuantity;
+import io.sphere.sdk.inventory.commands.updateactions.SetExpectedDelivery;
+import io.sphere.sdk.inventory.commands.updateactions.SetRestockableInDays;
+import io.sphere.sdk.inventory.commands.updateactions.SetSupplyChannel;
+import io.sphere.sdk.models.Reference;
+
+import javax.annotation.Nonnull;
+import java.time.ZonedDateTime;
+import java.util.List;
+import java.util.Objects;
+
+import static java.util.Arrays.asList;
+import static java.util.Collections.emptyList;
+
+/**
+ * This class provides static utility methods for building update actions related to inventories.
+ */
+public final class InventoryUpdateActionUtils {
+
+    private InventoryUpdateActionUtils() {
+        throw new AssertionError();
+    }
+
+    /**
+     * Compares the {@code quantityOnStock} values of a {@link InventoryEntry} and {@link InventoryEntryDraft}
+     * and returns a {@link List} of update actions, which would contain the {@code "changeQuantity"}
+     * {@link UpdateAction}. If quantity on stock from {@code newEntry} is {@code null} or no update action is needed,
+     * for example in case where both the {@link InventoryEntry} and the {@link InventoryEntryDraft} have the same
+     * {@code quantityOnStock} values, the empty list would be returned.
+     *
+     * @param oldEntry the inventory entry that should be updated
+     * @param newEntry the inventory entry draft which contains new quantity on stock.
+     * @return list containing update action or empty list if quantities on stock are indentical or quantity on stock
+     * from {@code newEntry} is null.
+     */
+    @Nonnull
+    public static List<UpdateAction<InventoryEntry>> buildChangeQuantityAction(@Nonnull final InventoryEntry oldEntry,
+                                                                               @Nonnull final InventoryEntryDraft newEntry) {
+        final Long oldValue = oldEntry.getQuantityOnStock();
+        final Long newValue = newEntry.getQuantityOnStock();
+        return newValue == null || Objects.equals(oldValue, newValue)
```

Comment:
@heshamMassoud It can be. Take a look at `InventoryEntryDraftBuilder`. Its public `of` methods take `quantityOnStock` as `Long` and use package private constructor from `InventoryEntryDraftDsl` during `build()`. Using such trick draft with `null` quantity on stock could be obtained.